torch/utils/cpp_extension.py

```diff
@@ -282,6 +282,25 @@ def _join_sycl_home(*paths) -> str:
     '-D__HIP_NO_HALF_CONVERSIONS__=1',
 ]
 
+_COMMON_SYCL_FLAGS = [
+    '-fsycl',
+    '-fsycl-targets=spir64_gen,spir64',
+]
+
+# Setting same default archs as in third_party/torch-xpu-ops project
+# minus dg2-* since it lacks hardware support for fp64 and requires
+# special consideration from the user. If needed this platform can
+# be requested thru TORCH_XPU_ARCH_LIST environment variable.
+_SYCL_ARCH_LIST = os.environ.get('TORCH_XPU_ARCH_LIST',
```

Comment:
Could we add a function to determine which arch list is preferred? When `TORCH_XPU_ARCH_LIST` is `None,` should we recommend using `torch.xpu.get_arch_list` by default?

Reply:
Done. `torch.xpu.get_arch_list` was added after I created a PR, so I missed it. I think it's a good idea to align with how pytorch is built using this mechanism. Thank you for advice.